conduit-combinators/Data/Conduit/Combinators.hs

```diff
@@ -1630,6 +1632,39 @@ slidingWindowC sz = go (max 1 sz) mempty
                        Just x -> go (n-1) (Seq.snoc st x)
 STREAMING(slidingWindow, slidingWindowC, slidingWindowS, sz)
 
+
+-- | Split input into chunk of size 'chunkSize'
+--
+-- The last element may be smaller than the 'chunkSize' (see also
+-- 'chunksOfExactlyE' which will not yield this last element)
+--
+-- @since 1.1.2
+chunksOfE :: (Monad m, Seq.IsSequence seq) => Seq.Index seq -> Conduit seq m seq
+chunksOfE chunkSize = chunksOfExactlyE chunkSize >> (await >>= maybe (return ()) yield)
+
+-- | Split input into chunk of size 'chunkSize'
+--
+-- If the input does not split into chunks exactly, the remainder will be
+-- leftover (see also 'chunksOfE')
+--
+-- @since 1.1.2
+chunksOfExactlyE :: (Monad m, Seq.IsSequence seq) => Seq.Index seq -> Conduit seq m seq
+chunksOfExactlyE chunkSize = await >>= maybe (return ()) start
+    where
+        start b
+            | onull b = chunksOfE chunkSize
```

Comment:
Small discrepancy, likely not worth changing: you have a special case here to deal with the null case, but below in `continue` there's no checking for null chunks.

Reply:
If I'm reading the code correctly, the behaviour would change in that, without this check, we could potentially call `leftover` with a null sequence. The check in `continue`, OTOH, would not change the behaviour at all.

Reply:
Far enough.